ApiDocs.DocumentationGeneration/DocumentationGenerator.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+ * Markdown Scanner
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved. 
+ * 
+ * MIT License
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of 
+ * this software and associated documentation files (the ""Software""), to deal in 
+ * the Software without restriction, including without limitation the rights to use, 
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so, 
+ * subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all 
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, 
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A 
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT 
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION 
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE 
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+namespace ApiDocs.DocumentationGeneration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Reflection;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    using ApiDocs.DocumentationGeneration.Extensions;
+    using ApiDocs.DocumentationGeneration.Properties;
+    using ApiDocs.Validation.OData;
+
+    using Mustache;
+
+    public class DocumentationGenerator
+    {
+        private readonly Generator resourceMarkDownGenerator;
+
+        public DocumentationGenerator()
+        {
+            string template = null;
+            using (var ms = new MemoryStream(Templates.resource))
```

Comment:
Instead of hard coding the template, could we allow it to be specified via command line parameter as well? Also, the name "resource" is a big generic, could you make this a more specific term?